src/server/models/listItem/providers/TranslatorsInterpreters.ts

```diff
@@ -6,61 +6,73 @@ import { logger } from "server/services/logger";
 import { prisma } from "server/models/db/prisma-client";
 import { fetchPublishedListItemQuery } from "server/models/listItem/providers/helpers";
 
-export async function findPublishedTranslatorsInterpretersPerCountry(props: {
+export interface FindPublishedTranslatorsInterpretersOptions {
   countryName?: string;
-  region?: string | "";
+  region?: string;
   servicesProvided?: string[];
   languagesProvided?: string[];
   interpreterServices?: string[];
   translationSpecialties?: string[];
   offset?: number;
-}): Promise<TranslatorInterpreterListItemGetObject[]> {
+}
+
+type Options = FindPublishedTranslatorsInterpretersOptions;
+
+export async function findPublishedTranslatorsInterpretersPerCountry(
+  props: Options
+): Promise<TranslatorInterpreterListItemGetObject[]> {
   if (props.countryName === undefined) {
     throw new Error("Country name is missing");
   }
-  const offset = props.offset ?? 0;
+
+  const {
+    region,
+    servicesProvided = ["translation", "interpretation"],
+    languagesProvided,
+    interpreterServices = [],
+    translationSpecialties = [],
+    offset = 0,
+  } = props;
+
   const countryName = startCase(toLower(props.countryName));
   const andWhere: string[] = [];
 
-  if (props.languagesProvided) {
+  if (languagesProvided) {
     andWhere.push(
       `AND ARRAY(select lower(jsonb_array_elements_text("ListItem"."jsonData"->'languagesProvided'))) && ARRAY ${JSON.stringify(
-        props.languagesProvided
+        languagesProvided
       ).replace(/"/g, "'")}`
     );
   }
-  if (props.servicesProvided && props.servicesProvided.length > 0 && !props.servicesProvided.includes("all")) {
+  if (servicesProvided) {
```

Comment:
There's probably something I've missed but how does this account for the `all` option being selected?

Reply:
there's no "all" option for this question - it just asks if the user wants a translator or interpreter as checkboxes.. so I'm not sure why there was an .includes("all") here to begin with

Reply:
Ah cool 👍🏿